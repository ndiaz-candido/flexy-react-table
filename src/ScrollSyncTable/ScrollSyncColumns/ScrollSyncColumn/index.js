import ScrollSyncColumn from './ScrollSyncColumn';

export default ScrollSyncColumn;
