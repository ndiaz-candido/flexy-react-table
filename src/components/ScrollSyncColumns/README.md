```js
<ScrollSyncTable>
  <ScrollSyncColumns>
    <ScrollSyncColumn name="a">Header A</ScrollSyncColumn>
    <ScrollSyncColumn name="b">Header B</ScrollSyncColumn>
  </ScrollSyncColumns>
  <ScrollSyncRows>
    <ScrollSyncRow>
      <ScrollSyncCell column="a">Cell for Column A</ScrollSyncCell>
      <ScrollSyncCell column="b">Cell for Column B</ScrollSyncCell>
    </ScrollSyncRow>
  </ScrollSyncRows>
</ScrollSyncTable>
```
