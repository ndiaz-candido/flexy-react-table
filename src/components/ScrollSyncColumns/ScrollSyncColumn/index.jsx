import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { scrollSyncColumn } from './index.module.css';

const ScrollSyncColumn = ({ width, children, className }) => {
  const styles = {
    width,
    flex: `0 0 ${width}`,
  };
  const classNames = classnames(className, scrollSyncColumn);

  return (
    <div className={classNames} style={styles}>
      {children}
    </div>
  );
};

ScrollSyncColumn.propTypes = {
  width: PropTypes.string,
  children: PropTypes.any.isRequired,

  /** The class name to override the default styles */
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

ScrollSyncColumn.defaultProps = {
  width: '150px',
  className: ''
};

export default ScrollSyncColumn;
