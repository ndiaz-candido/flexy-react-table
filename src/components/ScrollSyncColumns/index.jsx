import React from 'react';

const ScrollSyncColumns = () => <div />;

export default ScrollSyncColumns;
