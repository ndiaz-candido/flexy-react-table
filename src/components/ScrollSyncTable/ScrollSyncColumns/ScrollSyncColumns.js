import React from 'react';

const ScrollSyncColumns = () => (<div></div>);

export default ScrollSyncColumns;
