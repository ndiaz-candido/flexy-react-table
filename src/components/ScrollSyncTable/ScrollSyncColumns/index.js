import ScrollSyncColumns from './ScrollSyncColumns';

export default ScrollSyncColumns;
