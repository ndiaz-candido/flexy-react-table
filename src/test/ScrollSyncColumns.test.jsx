import React from 'react';
import { shallow } from 'enzyme';
import ScrollSyncColumns from '../components/ScrollSyncColumns';

describe('ScrollSyncColumns', () => {
  test('should render its default content', () => {
    const ScrollSyncColumnsComponent = shallow(<ScrollSyncColumns />);

    expect(ScrollSyncColumnsComponent).toMatchSnapshot();
  });
});
