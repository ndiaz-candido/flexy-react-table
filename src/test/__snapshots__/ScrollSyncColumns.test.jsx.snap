// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollSyncColumns should render its default content 1`] = `<div />`;
